fe/fe-core/src/main/java/com/starrocks/http/rest/ShowMetaInfoAction.java

```diff
@@ -179,27 +184,30 @@ public Map<String, Long> getDataSize() {
             List<Table> tables = GlobalStateMgr.getCurrentState().getLocalMetastore().getTables(db.getId());
             for (int j = 0; j < tables.size(); j++) {
                 Table table = tables.get(j);
-                if (table.getType() != TableType.OLAP) {
-                    continue;
+                if (table.isNativeTableOrMaterializedView()) {
+                    // in implementation, cloud native table is a subtype of olap table
+                    totalSize += calculateSizeForOlapTable((OlapTable) table, singleReplica);
                 }
-
-                OlapTable olapTable = (OlapTable) table;
-                long tableSize = 0;
-                for (PhysicalPartition partition : olapTable.getAllPhysicalPartitions()) {
-                    long partitionSize = 0;
-                    for (MaterializedIndex mIndex : partition.getMaterializedIndices(IndexExtState.VISIBLE)) {
-                        long indexSize = 0;
-                        for (Tablet tablet : mIndex.getTablets()) {
-                            indexSize += tablet.getDataSize(true);
-                        } // end for tablets
-                        partitionSize += indexSize;
-                    } // end for tables
-                    tableSize += partitionSize;
-                } // end for partitions
-                totalSize += tableSize;
-            } // end for tables
-            result.put(dbName, Long.valueOf(totalSize));
+            }
+            result.put(dbName, totalSize);
         } // end for dbs
         return result;
     }
+
+    @VisibleForTesting
+    public long calculateSizeForOlapTable(OlapTable olapTable, boolean singleReplica) {
+        long tableSize = 0;
+        for (PhysicalPartition partition : olapTable.getAllPhysicalPartitions()) {
+            long partitionSize = 0;
+            for (MaterializedIndex mIndex : partition.getMaterializedIndices(IndexExtState.VISIBLE)) {
```

Comment:
`MaterializedIndex` has the interface of `getDataSize`, can reuse the interface to support single or multiple replica size.

Reply:
Not to convenient though, `MaterializedIndex.getDataSize()` returns non-single replica size at the current implementation. Need to refactor it anyway.

Reply:
yes, you are right. However, the good side is, the dataSize() calculation are moved to MaterializedIndex and encapsulated as interface , others won't care about how the number is calculated.

In case the calculation need to be adjusted or bugfix needs to be applied, no one will be aware that the ShowMetaInfoAction.java needs to be adjusted too.